src/libcollections/string.rs

```diff
@@ -701,6 +701,13 @@ impl String {
     /// Violating these may cause problems like corrupting the allocator's
     /// internal datastructures.
     ///
+    /// The ownership of `ptr` is effectively transferred to the
+    /// `String` and the memory referred to by the pointer will be
+    /// deallocated when the `String` goes out of scope. Ensure that
```

Comment:
Should say “dropped” instead “goes out of scope”. Going out of scope does not necessarily imply the drop (we do not guarantee that, for example) and its the dropping which handles the deallocation.